client/my-sites/navigation/index.jsx

```diff
@@ -49,39 +46,13 @@ class MySitesNavigation extends Component {
 					{ ...asyncProps }
 				/>
 			);
-
-			// For the new Jetpack cloud sidebar, it has its own site picker.
-			renderSitePicker = false;
-
-			sitePickerProps = {
-				showManageSitesButton: false,
-				showHiddenSites: false,
-			};
 		} else if ( this.props.isGlobalSidebarVisible ) {
 			return this.renderGlobalSidebar();
 		} else {
 			asyncSidebar = <AsyncLoad require="calypso/my-sites/sidebar" { ...asyncProps } />;
-
-			sitePickerProps = {
-				showManageSitesButton: true,
-				showHiddenSites: true,
-				maxResults: 50,
-			};
 		}
 
-		return (
-			<div className="my-sites__navigation">
-				{ renderSitePicker && (
-					<SitePicker
-						allSitesPath={ this.props.allSitesPath }
```

Comment:
This is the only use of `allSitesPath` prop, so we could avoid passing it in?

https://github.com/Automattic/wp-calypso/blob/1bd19a9bac6361711a164078c0d7fdf09d68f2d0/client/my-sites/controller.js#L129

And it looks like that might have some cascading cleanup with the assignment of the variable.

Reply:
Seems like the `allSitesPath` prop is still used in a few places though:

https://github.com/Automattic/wp-calypso/blob/7024dff3f7503d27fe30ab921286270294707103/client/a8c-for-agencies/sections/sites/add-sites/atomic-sites-selector.tsx#L29

https://github.com/Automattic/wp-calypso/blob/7024dff3f7503d27fe30ab921286270294707103/client/jetpack-cloud/components/sidebar/site-selector.tsx#L58

So I'll keep it for now. 

Note that some other props can also be removed, but I'm planning to do further cleanup separately, as I also mentioned in the PR description:

> Note that I've got more removal coming up in https://github.com/Automattic/wp-calypso/pull/101844, but I'm landing it in digestible pieces.

Reply:
There still are `allSitesPath` props used across various components, but it's true that the one on `my-sites/navigation` can be removed. The `createNavigation` function `my-sites/controller` calculates the `allSitesPath` value and passes it to `MySitesNavigation`. This can be removed.

Reply:
Ah, I see what you two mean - cleaned up in https://github.com/Automattic/wp-calypso/pull/102364/commits/fc93a291b7c1ed9d7eeae322e2e97e60d6b7bcd0.

I'll suggest more cleanup in subsequent PRs.

Reply:
Yes, that's what I had in mind 🙂 Removing `SitePicker` apparently leads to an avalanche of further cleanups (like these in #101844), let's do them step by step.